### ISSUE TITLE
COBOL - Textmate and Cobol Language Server on "replacing" term in copy clause when last term of the line

### ISSUE THREAD

**Issue by FALLAI-Denis** [IBM/zopeneditor-about#346]
## Development environment used

- Z Open Editor version: 3.1.1
- Editor Platform
  - [X Visual Studio Code
  - [ ] Red Hat CodeReady Workspaces
  - [ ] Eclipse Che
  - [ ] Standalone Theia
- Editor Platform Version: 1.78.2
- Operating System: Windows 10
- Java Version: 11
- [ ] Related to RSE API?
  - RSE API Plugin version:
  - Zowe CLI version:
  - Node.js version:
- Logs attached: no

### Textmate error

The coloring of the term "replacing" in the "copy" directive depends on the content of the rest of the line:
- a sequel: the term "replacing" is well colored
- no continuation (no space): the term "replacing is badly colored

### COBOL Language Server error

If a floating comment is placed after the term "replacing" it is considered to be COBOL code.

### Observed behavior

![image](https://github.com/IBM/zopeneditor-about/assets/63861965/ab7b5717-29fe-42dd-bfc7-962d89ad1af3)

**Comment by FALLAI-Denis** [comment]
Related issue: #248

**Comment by FALLAI-Denis** [comment]
Hi,

For the textmate issue, I think the problem is here, and it's not just about the `replacing` keyword:

```
    {
      "match": "(?<![-_])(?i:...|replacing|...)(?=\\s|\\.|,)",
      "name": "keyword.identifers.cobol"
    },
```

the regular expression used expects a continuation after the keyword: a space, a point, or a comma.

It would also be necessary to provide the sequence end of line `$`.
Or maybe just indicate a word separator` \b`.

Test with end-of-line:

```
    {
      "match": "(?<![-_])(?i:...|replacing|...)(?=\\s|\\.|,|$)",
      "name": "keyword.identifers.cobol"
    },
```

![image](https://github.com/IBM/zopeneditor-about/assets/63861965/f698d145-3f9e-4091-9cb7-6e6b3b0fad3a)

This problem seems to be generalized in the textmate grammar because there are many expressions of the form `(?=\\s|\\.)` which do not provide for the physical end of line.

**Comment by FALLAI-Denis** [comment]
Hi,

Regarding regular expressions used to isolate words in a COBOL sentence, I suggest using the following pattern:

`(?<![\-])\b(?i:KEYWORD1|KEYWORD2|...|KEYWORDn)\b(?![\-])`

**Comment by phaumer** [comment]
Fixed in Z Open Editor v3.2.0
